foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/platform/database/OraclePlatform.java

```diff
@@ -661,7 +692,41 @@ protected ExpressionOperator logOperator() {
         result.bePrefix();
         result.setNodeClass(FunctionExpression.class);
         return result;
+    }
 
+    /**
+     * INTERNAL:
+     * Oracle equivalent to DATENAME is TO_CHAR.
+     */
+    protected static ExpressionOperator oracleDateName() {
+        ExpressionOperator exOperator = new ExpressionOperator();
+        exOperator.setType(ExpressionOperator.FunctionOperator);
+        exOperator.setSelector(ExpressionOperator.DateName);
+        Vector<String> v = org.eclipse.persistence.internal.helper.NonSynchronizedVector.<String>newInstance(3);
```

Comment:
As you are touching this, it would make sense to use `List<String>/ArrayList<>` instead of `Vector/NonSynchronizedVector<>` since `NonSynchronizedVector<>` is effectively an ArrayList implementing Vector...